In [2]:
import numpy as np
import pandas as pd

import seaborn as sns
import matplotlib.pyplot as plt

from scipy import stats
import statsmodels as sts 

plt.style.use('ggplot')


# Задания

Задачи, которые описаны исключительно текстом, из демонстрационного варианта НЭ по анализу данных.

### Д.И. для кол-ва пчёл

Винни-Пух в течение 150 дней фиксировал изменения количества пчёл в улье. Он уверен, что полученные наблюдения являются выборкой независимых одинаково распределённых нормальных случайных величин. Оказалось, что среднее количество пчёл равно 25000, а выборочная дисперсия равна 1300. Постройте 95%-ый доверительный интервал для математического ожидания количества пчёл в улье. 

### Время забега

In [2]:
df = pd.read_csv('../data/Running.csv')

# Переведем время в колонке MileMinDur в секунды и запишем в новую колонку
df['MileSecDur'] = df['MileMinDur'].apply(lambda x: int(x.split(':')[1])*60 + int(x.split(':')[2]))

df.head()

Athlete MileMinDur  MileSecDur
0        0    0:06:21         381
1        0    0:07:00         420
2        0    0:12:44         764
3        1    0:06:25         385
4        1    0:07:24         444

- **Athlete** - является студент спортсменом или нет
- **MileMinDur** - время, за которое пробегает студент 1 милю

На уровне значимости $\alpha=0.05$ протестируйте гипотезу о том, что спортстмены пробегают 1 милю быстрее. Посчитайте статистику, p-value и сделайте вывод.

### Количество осадков

Ниже приведены данные об уровнях осадков в двух различных регионах России, измеренные за одинаковые промежутки времени. Предполагая, что все необходимые предпосылки выполнены, дисперсии генеральных совокупностей равны, а выборки независимы, проверьте гипотезу о равенстве средних уровней осадков при помощи t-теста на уровне значимости 5%.

Регион 1: `[103.01,101.99,105.21,106.80,112.70,106.13,110.48,109.26,100.44,100.28]`.

Регион 2: `[107.38,106.31,106.00,105.27,105.27,104.66,103.70,105.07,105.12,104.74]`.

Посчитайте статистику, p-value и сделайте вывод.


In [3]:
region_1 = [103.01, 101.99, 105.21, 106.80, 112.70, 106.13, 110.48, 109.26, 100.44, 100.28]
region_2 = [107.38, 106.31, 106.00, 105.27, 105.27, 104.66, 103.70, 105.07, 105.12, 104.74]

### Корреляция

Выберите две переменные, между которыми возможно рассчитать интерпретируемый коэффициент выборочной корреляции, и вычислите этот коэффициент по приведённым данным. Ответ округлите до сотых.
Переменные:
- Цвет автомобиля, закодированный числами (1 - синий, 2 - красный, 3 - зелёный): `1, 1, 2, 3, 2]`
- Истинный объём бака автомобиля (л): `[50.1, 53.2, 55.0, 55.0, 50.2]`
- Пройденный километраж (тыс. км): `[15.2, 4.75, 1.2, 1.9, 9.3]`

Выберите верный вариант ответа
- (a)	0.21
- (b)	-0.59
- (c)	-0.41
- (d)	-0.94

На уровне значимости $\alpha=0.05$ дополнительно протестируйте гипотезу о равенстве нулю корреляции.

In [4]:
c = [1,1,2,3,2]
v = [50.1, 53.2, 55.0, 55.0, 50.2]
km = [15.2, 4.75, 1.2, 1.9, 9.3]

### heart (критерий Пирсона)

In [8]:
df = pd.read_csv('../data/heart.csv')

df['sex'] = df['sex'].map({1:'male', 0:'female'})
df['target'] = df['target'].map({1:'ill', 0:'healthy'})

df.head()

age     sex  cp  trestbps  chol  fbs  restecg  thalach  exang target
0   63    male   3       145   233    1        0      150      0    ill
1   37    male   2       130   250    0        1      187      0    ill
2   41  female   1       130   204    0        0      172      0    ill
3   56    male   1       120   236    0        1      178      0    ill
4   57  female   0       120   354    0        1      163      1    ill

- **sex** - категориальная переменная, является ли поступивший пациент мужчиной (кодируется 1) или женщиной (кодируется 0)
- **target** - было ли диагностировано сердечно-сосудистое заболевание у пациента или нет


На уровне значимости $\alpha=0.05$ протестируйте гипотезу о независимости этих двух переменных с помощью Критерия Пирсона и стандартного нормального распределения. Посчитайте статистики, p-value и сделайте вывод.

In [6]:
agg = df.pivot_table(index='sex', columns='target', values='age', aggfunc='count')

### Афросинья (критерий Пирсона)

Бабушка Афросинья хочет проверить, существует ли зависимость между сортом баклажана и типом почвы, в которую сорт был высажен. Для этого она собрала данные по 150 сортам, высаженных в чернозём, и 150 сортам, высаженных в каштановую почву. Данные представлены в таблице ниже.


| Почва/сорт | Алмаз | Матросик | Снежный |
|-|-|-|-|
| Чернозёмная почва | 75 | 20 | 55 |
| Каштановая почва | 20 | 120 | 10 |


На уровне значимости $\alpha=0.05$ протестируйте гипотезу о независимости этих двух переменных с помощью Критерия Пирсона. Посчитайте статистику, p-value и напишите вывод.

### Ошибки 1-2 рода

Соотнесите ситуацию и тип ошибки (I или II рода). Объясните свой выбор.

Ситуация 1: Студент получил оценку «неудовлетворительно» за взятый им курс. Предположим, что он формулировал нулевую гипотезу так: «Это достаточно простой курс, за который легко получить отличную оценку».

Ситуация 2: Элизабет вышла на прогулку без зонта, однако вскоре пошёл сильный дождь. Предположим, что она формулировала нулевую гипотезу так: «Сегодня будет дождь».